context/switch-arm64.c

```diff
@@ -0,0 +1,111 @@
+/* This file is part of the Diamond cothread library.
+ *
+ * Copyright (C) 2010-2012 Michael Abbott, Diamond Light Source Ltd.
+ *
+ * The Diamond cothread library is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * The Diamond cothread library is distributed in the hope that it will be
+ * useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * Contact:
+ *      Dr. Michael Abbott,
+ *      Diamond Light Source Ltd,
+ *      Diamond House,
+ *      Chilton,
+ *      Didcot,
+ *      Oxfordshire,
+ *      OX11 0DE
+ *      michael.abbott@diamond.ac.uk
+ */
+
+// Coroutine frame switching for ARMv8 64 bits
+
+#ifdef __ARM_NEON
+#define IF_NEON(code)    code
+#else
+#define IF_NEON(code)
+#endif
+
+__asm__(
+"       .text\n"
+
+// void * switch_frame(frame_t *old_frame, frame_t new_frame, void *arg)
+// Arguments on entry:
+//   x0      address of frame to be saved
+//   x1      frame to be loaded
+//   x2      Context argument to pass through
```

Comment:
Can you reference the relevant ABI documentation, please?  Have a look at `switch-x86.c` and `switch-x86_64.c` for examples of this.  I'd like to double check the registers you've chosen to save and restore

Reply:
Done, I added a bit explanation and a link to the [Procedure Call Standard for the Arm 64-bit Architecture document](https://github.com/ARM-software/abi-aa/releases/download/2021Q1/aapcs64.pdf)